From 0611c65b8e1c9c7c9f0f16c1542be904236c3fa7 Mon Sep 17 00:00:00 2001
From: William Lallemand <wlallem...@haproxy.com>
Date: Sun, 24 Jun 2018 09:37:03 +0200
Subject: [PATCH] BUG/BUILD: threads: unbreak build without threads

The build without threads was once again broken.

This issue was introduce in commit ba86c6c ("MINOR: threads: Be sure to
remove threads from all_threads_mask on exit").

This exactly the same problem that the last time it happened, because of
all_threads_mask not being defined with USE_THREAD=
---
 src/haproxy.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git src/haproxy.c src/haproxy.c
index 11d1d47ce..4101cefa1 100644
--- src/haproxy.c
+++ src/haproxy.c
@@ -2473,9 +2473,8 @@ static void *run_thread_poll_loop(void *data)
 	list_for_each_entry(ptdf, &per_thread_deinit_list, list)
 		ptdf->fct();
 
-	HA_ATOMIC_AND(&all_threads_mask, ~tid_bit);
-
 #ifdef USE_THREAD
+	HA_ATOMIC_AND(&all_threads_mask, ~tid_bit);
 	if (tid > 0)
 		pthread_exit(NULL);
 #endif
-- 
2.16.1
